src/nspawn/nspawn.c

```diff
@@ -1051,6 +1058,54 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_SYSCALL_FILTER: {
+                        bool invert = false;
+
+                        if (isempty(optarg)) {
+                                arg_syscall_filter = set_free(arg_syscall_filter);
+                        }
+
+                        if (optarg[0] == '~') {
+                                invert = true;
+                                optarg++;
+                        }
+
+                        if (!arg_syscall_filter) {
+                                arg_syscall_filter = set_new(NULL);
+                                if (!arg_syscall_filter)
+                                        return log_oom();
+
+                                if (invert)
+                                        /* Apply default filter and the ones listed */
```

Comment:
isn't this the other way around? if invert we're whitelisting so we're excluding the syscalls listed from the filter

Reply:
The comment was wrong 😛 

I was going by this https://www.freedesktop.org/software/systemd/man/systemd.exec.html#SystemCallFilter=
And there it acts as a whitelist by default and as a blacklist when you invert with `~`